lib/Core/Searcher.cpp

```diff
@@ -103,8 +103,17 @@ void BFSSearcher::update(ExecutionState *current,
   if (!addedStates.empty() && current &&
       std::find(removedStates.begin(), removedStates.end(), current) ==
           removedStates.end()) {
-    assert(states.front() == current);
-    states.pop_front();
+    if (states.front() == current) {
+      // BFS is the only searcher
+      states.pop_front();
+    } else {
+      // interleaved with other searcher
+      std::deque<ExecutionState*>::iterator pos = std::find(states.begin(),
```

Comment:
Can we reduce the code to this part of the branch, the `then` branch is just a case of the second one as well. Less overall code.